File: 프로그래머스/1/176963. 추억 점수/추억 점수.swift
import Foundation

func solution(_ name:[String], _ yearning:[Int], _ photo:[[String]]) -> [Int] {
    var value: [String:Int] = [:]
    var ans:[Int] = []
    for i in 0 ..< name.count {
        value[name[i]] = yearning[i]
    }
    print(value)
    
    for ph in photo {
        var sum = 0
        for i in ph {
            guard let val = value[i] else {
                continue
            }
            sum += val
        }
        ans.append(sum)
    }
    return ans
}

File: 프로그래머스/1/176963. 추억 점수/추억 점수.py
def solution(name, yearning, photo):
    answer = []
    dict = {}
    for i in range(len(name)):
        dict[name[i]] = yearning[i]
    
    for p in photo:
        total = 0
        for i in p:
            total += dict.get(i,0)
        answer.append(total)
        
    
    return answer